ansible_events_ui/ruleset.py

```diff
@@ -56,57 +70,85 @@ async def activate_rulesets(
-    for playbook in result.all():
-        with open(os.path.join(tempdir, playbook.name), "w") as f:
-            f.write(playbook.playbook)
-    rules_file = os.path.join(tempdir, "rules.yml")
-    with open(rules_file, "w") as f:
-        f.write(rulesets)
-    inventory_file = os.path.join(tempdir, "inventory.yml")
-    with open(inventory_file, "w") as f:
-        f.write(inventory)
-    vars_file = os.path.join(tempdir, "vars.yml")
-    with open(vars_file, "w") as f:
-        f.write(extravars)
+    settings = load_settings()
+    local_working_directory = working_directory
+    ensure_directory(local_working_directory)
+
+    if settings.deployment_type == "local":
+
+        # for local development this is better
+        cmd_args = [
+            ansible_events,
+            "--worker",
+            "--websocket-address",
+            "ws://localhost:8080/api/ws2",
+            "--id",
+            str(activation_id),
+        ]
+        logger.debug(ansible_events)
+        logger.debug(cmd_args)
+
+        proc = await asyncio.create_subprocess_exec(
+            ssh_agent,
+            *cmd_args,
+            cwd=local_working_directory,
+            stdout=asyncio.subprocess.PIPE,
+            stderr=asyncio.subprocess.STDOUT,
+        )
 
-    # initial version using docker
-    # mounting volumes is probably the wrong way to do this
-    # it would be better to build it into the container or pass it
-    # through a stream (stdin or socket)
-    # cmd = f"docker run -v {rules_file}:/rules.yml -v {inventory_file}:/inventory.yml -v {vars_file}:/vars.yml -it {execution_environment} ansible-events --rules /rules.yml -i /inventory.yml --vars /vars.yml"  # noqa
-
-    # try this with podman
-    # cmd = f"podman run -v {rules_file}:/rules.yml -v {inventory_file}:/inventory.yml -v {vars_file}:/vars.yml -it {execution_environment} ansible-events --rules /rules.yml -i /inventory.yml --vars /vars.yml"  # noqa
-
-    # for local development this is better
-    cmd = [
-        "--rules",
-        rules_file,
-        "-i",
-        inventory_file,
-        "--vars",
-        vars_file,
-        "--websocket-address",
-        "ws://localhost:8080/api/ws2",
-        "--id",
-        str(activation_id),
-    ]
-    logger.debug(ansible_events)
-    logger.debug(cmd)
-
-    proc = await asyncio.create_subprocess_exec(
-        ansible_events,
-        *cmd,
-        cwd=tempdir,
-        stdout=asyncio.subprocess.PIPE,
-        stderr=asyncio.subprocess.STDOUT,
-    )
+        activated_rulesets[activation_id] = proc
 
-    activated_rulesets[activation_id] = proc
+        task = asyncio.create_task(
+            read_output(proc, activation_id, db),
+            name=f"read_output {proc.pid}",
+        )
+        taskmanager.tasks.append(task)
+
+    elif (
+        settings.deployment_type == "docker"
+        or settings.deployment_type == "podman"
+    ):
+
+        docker = aiodocker.Docker()
```

Comment:
An external TCP endpoint for the docker socket?

Reply:
By default aiodocker will try to use the unix socket, which is the default setup for docker/podman. But it can be a TCP socket for external connections. 
Ref: 
https://docs.docker.com/engine/reference/commandline/dockerd/#daemon-socket-option
https://docs.podman.io/en/latest/markdown/podman-system-service.1.html

Nevermind, looking at aiodocker docs, the library supports `DOCKER_HOST` and `DOCKER_TLS_VERIFY` So it could be configured through these environment variables. Anyway it must be documented.